modular_skyrat/modules/delam_emergency_stop/code/scram.dm

```diff
@@ -0,0 +1,435 @@
+
+	// Restore the power that we were 'channelling' to the SM
+	addtimer(CALLBACK(SSnightshift, TYPE_PROC_REF(/datum/controller/subsystem/nightshift, restore_light_power)), rand(19 SECONDS, 21 SECONDS))
+
+/// Shatter the supermatter chamber windows
+/obj/structure/window/reinforced/plasma/proc/shatter_window()
+	visible_message(span_danger("The [src] shatters in the freon fire!"))
+	explosion(src, 0, 0, 0, 3, 5)
+	qdel(src)
+
+/// The valiant little machine falls apart, one time use only!
+/obj/machinery/atmospherics/components/unary/delam_scram/proc/goodbye_friends()
+
+	// good job buddy, sacrificing yourself for the greater good
+	playsound(src, 'sound/misc/compiler-failure.ogg', 80, FALSE, 20, ignore_walls = TRUE, use_reverb = TRUE)
+	audible_message(span_danger("The [src] beeps a sorrowful melody!"))
+	visible_message(span_danger("The [src] collapses into a pile of twisted metal and foam!"))
+	deconstruct(FALSE)
+
+/// Drain the internal energy, if the crystal damage is above 100 we heal it a bit. Not much, but should be good to let them recover.
+/obj/machinery/atmospherics/components/unary/delam_scram/proc/prevent_explosion()
+	var/obj/machinery/power/supermatter_crystal/engine/damaged_sm = my_sm?.resolve()
+	if(!damaged_sm)
+		return
+
+	damaged_sm.name = "partially delaminated supermatter crystal"
+	damaged_sm.desc = "This crystal has seen better days, the glow seems off and the shards look brittle. Central says it's still \"relatively safe.\" They'd never lie to us, right?"
+	damaged_sm.explosion_power = SM_DAMAGED_EXPLOSION_POWER // if you fuck up again, yeesh
+
+	if(damaged_sm.damage > SM_PREVENT_EXPLOSION_THRESHOLD)
+		damaged_sm.damage = SM_PREVENT_EXPLOSION_THRESHOLD
+
+	damaged_sm.internal_energy = MISTAKES_WERE_MADE
+	for(var/obj/machinery/power/energy_accumulator/tesla_coil/zappy_boi in range(3, src))
+		zappy_boi.stored_energy = MISTAKES_WERE_MADE
+
+/obj/machinery/atmospherics/components/unary/delam_scram/New()
+	. = ..()
+	var/datum/gas_mixture/delam_juice = new
+	delam_juice.add_gases(/datum/gas/freon)
+	delam_juice.gases[/datum/gas/freon][MOLES] = SM_COOLING_MIXTURE_MOLES
+	delam_juice.temperature = SM_COOLING_MIXTURE_TEMP
+	airs[1] = delam_juice
+
+/// Dims the lights and consumes a bit of APC power, summoning that electricity to stop the delam... somehow
+/datum/controller/subsystem/nightshift/proc/suck_light_power()
+	SSnightshift.can_fire = FALSE
+	for(var/obj/machinery/power/apc/light_to_suck as anything in SSmachines.get_machines_by_type(/obj/machinery/power/apc))
+		light_to_suck.cell.charge -= DELAM_MACHINE_POWER_CONSUMPTION
+		light_to_suck.lighting = APC_CHANNEL_OFF
+		light_to_suck.nightshift_lights = TRUE
+		light_to_suck.update_appearance()
+		light_to_suck.update()
+
+/// Restores the lighting after the delam suppression
+/datum/controller/subsystem/nightshift/proc/restore_light_power()
+	for(var/obj/machinery/power/apc/light_to_restore as anything in SSmachines.get_machines_by_type(/obj/machinery/power/apc))
```

Comment:
You need to set `SSnightshift.can_fire` back to `TRUE` in this proc, otherwise it's just going to stop working for the rest of the shift.

Reply:
Oop.